lib/Twig/Template.php

```diff
@@ -607,15 +618,33 @@ protected function getAttribute($object, $item, array $arguments = array(), $typ
             $this->env->getExtension('Twig_Extension_Sandbox')->checkMethodAllowed($object, $method);
         }
 
-        // Some objects throw exceptions when they have __call, and the method we try
-        // to call is not supported. If ignoreStrictCheck is true, we should return null.
-        try {
-            $ret = call_user_func_array(array($object, $method), $arguments);
-        } catch (BadMethodCallException $e) {
-            if ($call && ($ignoreStrictCheck || !$this->env->isStrictVariables())) {
-                return;
+        if ($call) {
+            // Some objects throw exceptions when they have __call, and the method we try
+            // to call is not supported. If ignoreStrictCheck is true, we should return null.
+            try {
+                $ret = $object->__call($method, $arguments);
+            } catch (BadMethodCallException $e) {
+                if ($ignoreStrictCheck || !$this->env->isStrictVariables()) {
+                    return;
+                }
+                throw $e;
             }
-            throw $e;
+        } elseif ($a = $arguments) {
+            switch (count($a)) {
+                case 1: $ret = $object->$method($a[0]); break;
```

Comment:
BC break removed as far as public interfaces are concerned: there is now no way to call this method with an associative array by writing twig code.
Since the method is protected, there is a theoretical possibility for a class extending `Twig_Template` and calling it with an associative array. But not a Twig generated one.
But seriously, such a narrow case should not be taken as a blocker here IMHO.